src/ConfigUpdater.php

```diff
@@ -0,0 +1,279 @@
+        $entity->save();
+      }
+
+      $succeeded[$config_name] = $this->t('@config was successfully reverted', ['@config' => $config_name]);
+    }
+
+    return new ConfigUpdaterResult($failed, $succeeded);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function delete(array $config_names) {
+    $succeeded = [];
+    $failed = [];
+
+    foreach ($config_names as $config_name) {
+      $value = $this->loadFromActive($config_name);
+      if (!$value) {
+        $failed[$config_name] = $this->t('@config did not exist in extension storage', ['@config' => $config_name]);
+        continue;
+      }
+
+      // Check the configuration object's hash and see if it has been modified.
+      if ($this->isModified($config_name)) {
+        $failed[$config_name] = $this->t('@config has been modified and was not deleted', ['@config' => $config_name]);
+        continue;
+      }
+
+      $type = $this->getConfigType($config_name);
+      if ($type == 'system.simple') {
+        $config = $this->configFactory->getEditable($config_name);
+        if (!$config) {
+          $failed[$config_name] = $this->t('@config did not exist in active storage', ['@config' => $config_name]);
+          continue;
+        }
+        $config->delete();
+      }
+      else {
+        $definition = $this->entityTypeManager->getDefinition($type);
+        $id_key = $definition->getKey('id');
+        $id = $value[$id_key];
+
+        /** @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $entity_storage */
+        $entity_storage = $this->entityTypeManager->getStorage($type);
+        $entity = $entity_storage->load($id);
+        $entity_storage->delete([$entity]);
+      }
+
+      $succeeded[$config_name] = $this->t('@config was successfully deleted', ['@config' => $config_name]);
+    }
+
+    return new ConfigUpdaterResult($failed, $succeeded);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function loadFromActive($config_name) {
+    return $this->activeConfigStorage->read($config_name);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function loadFromExtension($config_name) {
+    $value = $this->extensionConfigStorage->read($config_name);
+    if (!$value) {
+      $value = $this->extensionOptionalConfigStorage->read($config_name);
+    }
+    return $value;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isModified($config_name) {
+    // Read the configuration from active storage.
+    $active = $this->activeConfigStorage->read($config_name);
+
+    // Get the hash created when the config was installed.
+    $original_hash = $active['_core']['default_config_hash'];
+
+    // Remove export keys not used to generate default config hash.
+    unset($active['uuid']);
+    unset($active['_core']);
+
+    // Recalculate the hash.
+    $active_hash = Crypt::hashBase64(serialize($active));
+
+    return $original_hash !== $active_hash;
+  }
+
+  /**
+   * Returns the config type for a given config object.
```

Comment:
"Gets the config type for a given config object." -> the verb in the docblock should match the verb in the method name.